modules/pam_tty_audit/pam_tty_audit.c

```diff
@@ -351,6 +351,14 @@ pam_sm_open_session (pam_handle_t *pamh, int flags, int argc, const char **argv)
 
   fd = nl_open ();
   if (fd == -1
+      && errno == EPROTONOSUPPORT)
+    {
+      pam_syslog (pamh, LOG_WARNING, "unable to open audit socket, audit not "
+                  "supported; tty_audit skipped");
+      free (old_status);
+      return PAM_SUCCESS;
```

Comment:
Thinking about it more. Please change the return value to PAM_IGNORE in this case. This way it will behave like PAM_SUCCESS with required and so but it will be possible to distinguish the error from other errors or success.

Reply:
Done